packages/core/src/ceramic.ts

```diff
@@ -331,19 +344,7 @@ class Ceramic implements CeramicApi {
    * @private
    */
   private _getDocFromCache(docId: DocID): Document {
-    return this._docmap[docId.toString()]
-  }
-
-  /**
-   * Puts document into cache if not already there
-   * @param doc - document to cache
-   * @private
-   */
-  private _cacheDocIfNeeded(doc: Document) {
-    const docIdStr = doc.id.toString()
-    if (!this._docmap[docIdStr]) {
-      this._docmap[docIdStr] = doc
-    }
+    return this._docCache.get(docId) as Document
```

Comment:
The DocCache class only caches the DocStateHolder type. I assume we need to create an instance of the document here?

Reply:
hm, `Document` also implements `DocStateHolder`

Reply:
But will `DocStateHolder as Document` invoke the `Document` constructor?  I don't think it does, which means that other members of `Document` such as `_logger`, `_applyQueue`, `_genesisCid`, etc won't be properly initialized

Reply:
Why do we need to invoke the constructor?

Reply:
Wrote that because I thought the cache just had the state, so we needed to construct a real Document instance.  Now I understand that the cache actually does store full Document objects, so this `as` is just a cast to make the type checker happy